### ISSUE TITLE
Bitwarden CLI takes an extremely long time to execute on macOS

### ISSUE THREAD

**Issue by jabofh** [bitwarden/clients#7742]
### Steps To Reproduce

``` console
eroux@homebase:~$ time bw list items --session "$BWSESSIONID" > log.txt # Using the latest Homebrew package
(node:31859) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
bw list items --session  > log.txt  101.38s user 23.94s system 139% cpu 1:30.14 total
eroux@homebase:~$ time ./bw list items --session "$BWSESSIONID" > bw.log # Using the latest binary from https://bitwarden.com
./bw list items --session  > bw.log  147.98s user 35.52s system 143% cpu 2:08.05 total
eroux@homebase:~$ 
```

### Expected Result

I expect a return in the same time it does with the Browser Plugin and GUI app, a maximum of a couple of seconds.

### Actual Result

As you can see in the "Steps To Reproduce" section, it takes, at quickest, 90 seconds. I have had it take nearly 4 minutes on more than one occasion.

Also note that this seems to be limited to macOS only. Running the same command on Linux I got a complete response in less than 5 seconds.

### Operating System

macOS

### Operating System Version

Sonoma 14.3

### Shell

Zsh

### Build Version

2024.1.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by daniellbw** [comment]
Closed as this is a duplicate of #7133

**Comment by watzkuh** [comment]
Hello!

I hope it's okay if re-use this closed issue.

I just wanted to report that the issue still persists for me, my setup is exactly the same: `bw list items` takes >90 seconds. -- Build 2024.1.0 on MacOs Sonoma 14.3

It might be worth to mention that the issue correlates with the number of items that are returned by the call (> 1500 in my case). I tracked down the issue to [this call](https://github.com/bitwarden/clients/blob/337176077957ebcd50c94075fd71ce3fd2c855a7/libs/common/src/platform/services/state.service.ts#L2599C11-L2599C35) which is needlessly executed for every item.
It seems to me like this would be trivial to fix, I prepared a PR for that: https://github.com/bitwarden/clients/pull/7855
